---
title: "ChIP-Seq -- ChIP-seq annotation and visualization --"
author: "Denis Puthier"
date: '`r Sys.Date()`'
output:
  html_document:
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    css: style.css
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  word_document: default
---


## ChIP-seq annotation and visualization


### Global Objective
Given a set of ChIP-seq peaks annotate them in order to find associated genes, genomic categories and functional terms.

### Data set
For this practical session, the ChIP-seq data and peaks related to following publication will be used: “GATA3 acts upstream of FOXA1 in mediating ESR1 binding by shaping enhancer accessibility.”, [Theodorou _et al_](http://www.ncbi.nlm.nih.gov/pubmed/23172872).


The following pre-processed datasets are available:

| Sample						| format |
|-------------------|--------|
| [siGATA_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r1_SRX176857_peaks.bed) | bed|
| [siGATA_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r2_SRX176859_peaks.bed) | bed|
| [siGATA_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r3_SRX176861_peaks.bed) | bed|
| [siGATA_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_H3K4me1_E2_r1_SRX176879_broad_peaks.bed) | bed|
| [siNT_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r1_SRX176856_peaks.bed) | bed|
| [siNT_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r2_SRX176858_peaks.bed) | bed|
| [siNT_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r3_SRX176860_peaks.bed) | bed|


| Sample						| format |
|-------------------|--------|
| [MCF_input_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/MCF_input_r3_SRX176888_sort.bam) | bam|
| [siGATA_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r1_SRX176857_sort.bam) | bam|
| [siGATA_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r2_SRX176859_sort.bam) | bam|
| [siGATA_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r3_SRX176861_sort.bam) | bam|
| [siGATA_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_H3K4me1_E2_r1_SRX176879_sort.bam) | bam|
| [siNT_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r1_SRX176856_sort.bam) | bam|
| [siNT_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r2_SRX176858_sort.bam) | bam|
| [siNT_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r3_SRX176860_sort.bam) | bam|
| [siNT_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_H3K4me1_E2_r1_SRX176878_sort.bam) | bam|



| Sample						| format |
|-------------------|--------|
| [MCF_input_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/MCF_input_r3_SRX176888_sort.bam.bai) | bam|
| [siGATA_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r1_SRX176857_sort.bam.bai) | bam|
| [siGATA_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r2_SRX176859_sort.bam.bai) | bam|
| [siGATA_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r3_SRX176861_sort.bam.bai) | bam|
| [siGATA_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_H3K4me1_E2_r1_SRX176879_sort.bam.bai) | bam|
| [siNT_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r1_SRX176856_sort.bam.bai) | bam|
| [siNT_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r2_SRX176858_sort.bam.bai) | bam|
| [siNT_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r3_SRX176860_sort.bam.bai) | bam|
| [siNT_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_H3K4me1_E2_r1_SRX176878_sort.bam.bai) | bam|


| Sample						| format |
|-------------------|--------|
| [MCF_input_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/MCF_input_r3_SRX176888.bigWig) | bigWig|
| [siGATA_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r1_SRX176857.bigWig) | bigWig|
| [siGATA_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r2_SRX176859.bigWig) | bigWig|
| [siGATA_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_ER_E2_r3_SRX176861.bigWig) | bigWig|
| [siGATA_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siGATA_H3K4me1_E2_r1_SRX176879.bigWig) | bigWig|
| [siNT_ER_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r1_SRX176856.bigWig) | bigWig|
| [siNT_ER_E2_r2](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r2_SRX176858.bigWig) | bigWig|
| [siNT_ER_E2_r3](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_ER_E2_r3_SRX176860.bigWig) | bigWig|
| [siNT_H3K4me1_E2_r1](http://pedagogix-tagc.univ-mrs.fr/courses/data/ngs/td_chip_seq/all/siNT_H3K4me1_E2_r1_SRX176878.bigWig) | bigWig|


## Loading the data

Reload IGV. Load the bigWig files and bed files in the genome browser. Use colored tracks to distinguish between GATA3 KO and WT condition. Organize properly the files with peaks below corresponding bigwigs.

<div class="exo">
 - What about the H3K4me1 track ? Does it looks like ESR1 track ?
 - Using galaxy search for the most enriched peak relative to input (best score) for one ESR1 sample and one H3K4me1 sample.
 - Using galaxy, draw the distribution of peak size for ESR1 and  H3K4me1. Do they look the same.
 - For each bed the 7th column contains the summit position relative to the start position. For one sample convert it to a bed format and load it into IGV.
 - Extend the summit position by 100 nucleotides in 5' and 3' using bedtools. Laod the result in IGV.
</div>


## Comparing replicates

Reproducibility may be an issue in ChIP-Seq experiment. Select one condition (WT or GATA KO).

<div class="exo">

- What are the number of peaks in each replicate.
- Using intersectBed from the Bedtools suite (available in Galaxy or command line), compare the peaks shared by the 3 replicates (WT or KO). You could draw a venn diagram with this information. 
- Create a bed file containing peaks that overlap in all tree replicates. 
</div>

## Relate peaks to GO terms

For that specific step we will use the GREAT annotation tools. Connect to [GREAT web server](http://great.stanford.edu) and perform a GO annotation for the ESR1 peaks. Alternatively GREAT can be launch directly from UCSC web server (using Table browser Custom track and by selecting send to GREAT). 

<div class="protocol">

> - Connect the [GREAT](http://great.stanford.edu) web server [http://great.stanford.edu](http://great.stanford.edu)
> - select the genome assembly version (hg19)
> - Upload or paste the peaks obtained previously in BED format (e.g intersect from the 3 replicates).
> - use the whole genome as background and run the software
> - Try using other parameters ("marked" regions). What about the results ?
> - Check the result with a negative control (use randomBed in Galaxy to produce random regions).

</div> 

<div class="exo">
Examine the enriched functional categories ? Are they biologically meaningful given what you know about ESR1 ?
</div>


### Using CEAS

In the TAGC galaxy instance. CEAS (Cis-regulatory Element Annotation System) is a tool for characterizing genome-wide protein-DNA interaction patterns from ChIP-chip and ChIP-Seq of both sharp and broad binding factors. It provides statistics on ChIP enrichment at important genome features such as specific chromosome, promoters, gene bodies, or exons, and infers genes most likely to be regulated by a binding factor.

<div class="protocol">

> - Select **NGS: Peak Calling > CEAS Annotate intervals and scores with genome features**.
> - Use the appropriate settings and click **Execute**.

</div>

What can we  say about the enrichment.


## Visualize ChIP enrichment around a given feature

Using the “deepTools heatmapper” we will try to visualize the local enrichment around the TSS for all known genes. Before drawing the heatmap we need to prepare the data by computing a summary matrix of the  local ChIP enrichment using “deepTools computeMatrix”.

<div class="protocol">

> - Download the required annotation file (here all UCSC annotated genes) from the UCSC genome browser (go to Table Browser, UCSC Genes and download the annotation as bed file. Save this locally). 
> - Use the obtained annotation file and the previously computed bigWig profile (from your history) as an input to “computeMatrix”. You could select for instance H3K4me1.
> - Use 'reference point’ as  the output option and 'beginning of region’ as this reference point (TSS).
> - Using the “heatmapper”, load the obtained matrix data and fill the desired options to plot the heat map.
> - Perform the analysis both with a ChIP for a chromatin mark and a transcription factor.

</div>

<div class="exo">
Do you see any differences between the two choosen dataset ?
</div>


## Integrative ChIP-seq analysis of regulatory elements

In this part, we will use the ReMap (http://tagc.univ-mrs.fr/remap/index.php) software to compare the peaks obtained in the peaks calling tutorial to an extensive regulatory catalog of 8 million transcription factor binding regions defined by collecting all the peaks from ChIP-seq experiments from the ENCODE project + other public datasets from the GEO database. Note that on the ReMap Web site, the term "site" is used to denote a ChIP-seq peak, rather than the precise binding location of a transcription factor

<div class="protocol">

> - Connect the [ReMap](http://tagc.univ-mrs.fr/remap/index.php) web server
> - Go to the Annotation Tool
> - upload or paste the peaks in BED format (select BED format in the data format selector)
> - Add your email and run the software with default parameters

</div> 

<div class="exo">
What are the TFs associated to your peaks?
</div>










