---
title: "Structure-based analysis of transcriptional regulation"
author:
- affiliation: Fundación ARAID & Estación Experimental de Aula Dei-CSIC, Zaragoza,
    España <http://www.eead.csic.es/compbio> | chercheur invité AMU, Marseille, France
  name: Bruno Contreras Moreira
date: "October, 2015"
output:
  html_document:
    fig_caption: yes
    toc: yes
  pdf_document:
    fig_caption: yes
    toc: yes
bibliography: bib_myarticles_md.bib
<!-- csl: biomed-central.csl , por defecto chicago bib style-->
---

___

## Protein-DNA recognition

DNA-binding proteins contain DNA-binding domains and have a specific or general affinity for either single or double stranded DNA. Here we will concentrate mostly on transcription factors, which generally recognize cis-regulatory elements in double-stranded DNA molecules.

### Dissecting a protein-DNA interface

Transcription factors recognize target DNA sequences through a binding interface, composed of protein residues and DNA stretches in intimate contact.
The best descriptions of protein-DNA interfaces are provided by structural biology, usually by X-ray or NMR experiments.

![Details of the interface of Lac repressor and its operator site, @Lewis1996](pics/Lac-1.png) 

![deatils of the interface of Lac repressor and its operator site, @Lewis1996](pics/Lac-2.png) 

![Cartoon dissection of the interface of a NarL dimer (PDB entry 1JE8)](pics/interface_movie.gif) 

### Atomic interactions between protein and DNA residues

The process of recognition of DNA sequences by proteins involves readout mechanisms, and also accessory stabilizing atomic interaction that do not confer specificty.

**Direct readout** 

* Hydrogen bonds: direct + water-mediated
* Hydrophobic interactions

**Indirect readout**

* Sequence-specific deformation of DNA base steps

**Stabilizing interactions**

* Not sequence-specific, involving DNA backbone

### Direct readout: hydrogen bonds

![Donor and acceptor hydrogen bond atoms in nitrogen bases, from @Luscombe2001](pics/bases-groove-HB.jpg) 

### Direct readout: Van der Waals interactions

![DNA bases display VdW interactions involving mainly C atoms, from @Luscombe2001](pics/bases-VW.gif)

![Sequence-specific VdW interactions involve fundamentally thymine-C5M group, adapted from @Luscombe2001](pics/bases-groove-VW.jpg)

### Indirect/shape readout

Besides atomic interactions between protein and DNA, Sequence-dependent deformability of duplexes, deduced from crystal complexes, implies that sequence recognition also involves DNA shape.

![DNA step geometry descriptors, from @Lu2008](pics/bp_step_hel.png)

DNA deformation is described by the increase in energy brought about by instantaneous fluctuations of the step parameters from their equilibrium values:

$deformation = \displaystyle\sum_{i=0}^6 \displaystyle\sum_{j=0}^6 spring_{ij} \Delta\theta_{i,st} \Delta\theta_{j,st}$ @Olson1998

![Roll and twist of DNA steps are different for purine and pyrimidine steps, from @Olson1998](pics/roll-twist-steps.jpg)

The accumulation of experimental and structural and molecular dynamics simulations

![Prediction of geometry descriptors of a DNA duplex by sliding a sequence window, from @Zhou2013](pics/DNApentamers.jpg)

### Protein-DNA interface graphs

Interfaces have been can be explored as generic bipartite graphs:

![Generic graph representation of the interface among protein residues (yellow) and DNA bases (blue), from @Sathyapriya2008](pics/bipartite-graph.png)

Or sub-graphs that focus on specific sequence recognition:

![Interface graph of NarL, where solid bases indicate indirect readout and arrows direct readout interactions, from @ContrerasMoreira2010. Adapted from <http://floresta.eead.csic.es/3dfootprint/complexes/1je8_F.html> ](pics/1je8-intf.png)



### Comparison of DNA-binding proteins

The literature describes a great variety of natural DNA-binding proteins, which can be analyzed and compared in terms of the features introduced above, such as readout, or instead with an evolutionary or topological perspective.  

![Example HTH transcription factors from the work of @Luscombe2000](pics/DNAbinding-classes.jpg)

The  [Structural Classification of Proteins (SCOP)](http://scop.berkeley.edu) systematically groups protein folds in superfamilies, of which some are the most common DNA-binding proteins. The next table shows superfamilies with more than 20 non-redundant complexes in the [Protein Data Bank](http://www.rcsb.org) as of October, 2015, as annotated in the database [3d-footprint](http://floresta.eead.csic.es/3dfootprint/):

SCOP superfamily | Number of complexes
---------------- | -------------------
Winged helix (WH) | 77
Homeodomain-like (H) | 63
Glucocorticoid-receptor-like (GR) | 33
Restriction endonuclease-like (RE) | 24 
Homing endonuclease (HE) | 23 
p53-like (P53) | 21
Lambda-repressor-like (LR) | 21

![Number of hydrogen bonds and VdW interactions across some SCOP superfamilies of DNA-binding proteins (right) and total atomic contacts per monomer (left). ZF=C2H2 Zinc Finger, RHH=Ribbon-helix-helix, from @ContrerasMoreira2010b](pics/direct-stats.png)

![Frequency of distorted DNA steps (indirect readout) of target DNA sites of some SCOP superfamilies of DNA-binding proteins, from @ContrerasMoreira2010b](pics/indirect-stats.png)

### Family wide analysis of protein-DNA interfaces

The available experimental structures of protein-DNA complexes in the PDB support the annotation of interface residues, those involved directly in sequence recognition, within protein families.

![Annotation of interface residues upon dissection of PDB complexes and their distribution, from @Sebastian2014](pics/intf-annot-hist.png)

ejemplos 

![Relation between (annotated) interface residues and cognate DNA motifs or homeobox transcription factors, from @Noyes2008](pics/homeo-intf.png)

### Exercise

Ejercicio con TFcompare con TFs del paper
http://161.111.227.80/compbio/material/regulatory_proteins/node6.html

Homology prediction of binding interfaces and readout
BLAST, AI
Ejercicio con 3d-footprint

___

## Mining sequence motifs from the structure of protein-DNA complexes

### Total contacts model
Morozov, Lichtarge

### Readout model
DNAPROT, Sarai, Lavery

### Specificity-based classification of DNA-binding proteins
DNAPROT

### Genomic benchmarks 

Ejercicio con 3d-footprint & matrix-quality, con alugnp por ejemplo
http://161.111.227.80/compbio/material/regulatory_proteins/node7.html

### 3D-footprint and footprintDB
a repository of DNA motifs, including structure-based, which annotates the corresponding binding protein and its interface

Ejercicio con footprintDB (capítulo de libro?)

___

## References
